File: csharp/parallel-letter-frequency/ParallelLetterFrequency.cs
using System.Collections.Generic;
using System.Linq;

public static class ParallelLetterFrequency
{
    public static Dictionary<char, int> Calculate(IEnumerable<string> texts) =>
        texts
            .AsParallel()
            .SelectMany(text => text.ToLower().Select(c => c))
            .Where(char.IsLetter)
            .GroupBy(c => c)
            .ToDictionary(c => c.Key, c => c.Count());
}

File: vbnet/parallel-letter-frequency/ParallelLetterFrequency.vb
Imports System.Collections.Generic
Public Module ParallelLetterFrequency
    Public Function Calculate(ByVal texts As IEnumerable(Of String)) As Dictionary(Of Char, Integer)
        Return texts.AsParallel().SelectMany(Function(s) s.ToLower().Select(Function(c) c)).Where(Function(c) Char.IsLetter(c)).GroupBy(Function(c) c).ToDictionary(Function(c) c.Key, Function(c) c.Count())
    End Function
End Module
